### ISSUE TITLE
Spell checking support

### ISSUE THREAD

**Issue by aperezdc** [aperezdc/revolt#61]
WebKitGTK+ provides spell checking support for HTML forms, which would make spell-checking work for Riot. We need:

* [Enabling the spell checking in the WebContext](https://webkitgtk.org/reference/webkit2gtk/stable/WebKitWebContext.html#webkit-web-context-set-spell-checking-enabled).
* [Setting a list of enabled languages](https://webkitgtk.org/reference/webkit2gtk/stable/WebKitWebContext.html#webkit-web-context-set-spell-checking-languages).
* Provide a widget in the settings window to allow the user choosing languages.

**Comment by thanostx** [comment]
To save you some time, there is a PR upstream that adds spell checking on riot.

https://github.com/vector-im/riot-web/pull/4215

**Comment by MTRNord** [comment]
As I am the one doing that PR: I am currently traped to work on it as Windows 10 currently is buggy because of a upstream issue so I can't do anything else than wait until they resolved the upstream issue

**Comment by MTRNord** [comment]
PS: the upstream spellcheck support will be electron only as it isn't a website feature. It is a browser feature which in that case will be electron

**Comment by aperezdc** [comment]
Yes, @MTRNord is right indeed. Spell-checking support depends on the Web browser or, in our case here, in WebKitGTK+'s spell-checking support, which we need to enable as per the bug description :wink:

**Comment by indolering** [comment]
Please make sure that the spell checker doesn't leak plaintext, like through an online service.  If possible, isolate custom entries to Revolt - as it introduces a side channel other apps can pick up on.
